### PR TITLE
Feat: add op.#Suspend and suspending phase in step

### PR DESCRIPTION
**Pull request by FogDong**
### Description of your changes

This PR upgrades the workflow version and adds op.#Suspend,  suspending phase in step.

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### Special notes for your reviewer

cc @barnettZQG

### CODE REVIEW

File: charts/vela-core/templates/defwithtemplate/suspend.yaml [hunk 1]
```diff
@@ -12,8 +12,20 @@ spec:
...
+        	if parameter.duration != _|_ {
+        		duration: parameter.duration
+        	}
+        	if parameter.message != _|_ {
+        		message: parameter.message
+        	}
+        }
         parameter: {
-        	// +usage=Specify the wait duration time to resume workflow such as "30s", "1min" or "2m15s"
         	duration?: string
```
Comment (StevenLeiZhang) [inline charts/vela-core/templates/defwithtemplate/suspend.yaml:28]: does ·duration· need default value？

File: pkg/workflow/operation/operation.go [hunk 2]
```diff
@@ -83,19 +85,70 @@ func (wo appWorkflowOperator) Suspend(ctx context.Context) error {
...
+				wfUtils.OperateSteps(steps, i, -1, workflowv1alpha1.WorkflowStepPhaseSuspending)
+				found = true
+				break
+			}
 		}
-		// set the workflow suspend to true
-		app.Status.Workflow.Suspend = true
-		return wo.cli.Status().Patch(ctx, app, client.Merge)
+		for j, sub := range step.SubStepsStatus {
+			if sub.Phase == v1alpha1.WorkflowStepPhaseRunning {
```
Comment (StevenLeiZhang) [inline pkg/workflow/operation/operation.go:130]:
```
if sub.Phase != v1alpha1.WorkflowStepPhaseRunning {
   continue
}

// your logic here
```

you will get less indent

File: pkg/workflow/operation/operation.go [hunk 3]
```diff
@@ -83,19 +85,70 @@ func (wo appWorkflowOperator) Suspend(ctx context.Context) error {
...
+					break
+				}
+			}
+		}
+	}
+	if !found {
+		return fmt.Errorf("can not find step %s", stepName)
+	}
+	b, _ := json.Marshal(app.Status.Workflow)
+	fmt.Println(string(b))
```
Comment (StevenLeiZhang) [inline pkg/workflow/operation/operation.go:145]: is it logger?

File: pkg/workflow/operation/operation.go (hunk 2 above)
Comment (FogDong) [reply pkg/workflow/operation/operation.go:130]: Nice catch!

File: charts/vela-core/templates/defwithtemplate/suspend.yaml (hunk 1 above)
Comment (FogDong) [reply charts/vela-core/templates/defwithtemplate/suspend.yaml:28]: I don't think duration needs a default value, it should be decided by the user.

Comment (barnettZQG) [reply charts/vela-core/templates/defwithtemplate/suspend.yaml:28]: Why did we remove the description?

**Review by barnettZQG — APPROVED** [review]
LGTM

**Review by Somefive — APPROVED** [review]

**Review by chivalryq — APPROVED** [review]
LGTM

**Pull request merged**